portal-impl/src/com/liferay/portal/util/HttpImpl.java

```diff
@@ -1227,6 +1227,15 @@ public String shortenURL(String url, int count) {
 
 		StringBundler sb = new StringBundler();
 
+		String[] parts = StringUtil.split(url, CharPool.QUESTION);
+
+		if (parts.length == 2) {
```

Comment:
You don't know how many "?" do you have in the incoming URL, in case there is more than 1, this logic will be completely broken.